File: services/providerDefinitions/ProviderDataContracts/Filters/DimensionFilter.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace ProviderDataContracts.Filters
{
    [DataContract]
    public class DimensionFilter
    {
        [DataMember(EmitDefaultValue = false, Name = "dimensionId")] public string DimensionID { get; set; }
        [DataMember(EmitDefaultValue = false)] public Object ServerContextData { get; set; }
        [DataMember(EmitDefaultValue = false, Name="attributes")] public IEnumerable<string> AttributeIDs { get; set; }

        public override bool Equals(object obj)
        {
            DimensionFilter filter = obj as DimensionFilter;
            if (filter == null) return false;

            return string.Equals(DimensionID, filter.DimensionID) &&
                object.Equals(ServerContextData, filter.ServerContextData) &&
                Enumerable.SequenceEqual<string>(AttributeIDs, filter.AttributeIDs);
        }

        public override int GetHashCode()
        {
            return DimensionID.GetHashCode();
        }
    }
}

File: scripts/statistics/Model/Filter/DimensionFilter.js

Statistics.Model.Filter = {};
Statistics.Model.Filter.DimensionFilter = Statistics.Class({
	
	/**
	 * @public
	 * @property {string}
	 */
	dimensionId: null,

	/**
	 * @public
	 * @property {string[]}
	 */	
	attributeIds: null,
	
	/**
	 * @constructor
	 * @param {string} id
	 * @param {string[]} attrids
	 */
	_init: function(id, attrids) {
		this.dimensionId = id;
		this.attributeIds = attrids;
	}
});


Statistics.Model.Filter.DimensionFilter.FromObject = function(obj){
	
	return new Statistics.Model.Filter.DimensionFilter(obj.dimensionId, obj.attributes);
}
